chore: OPTIC-1632: Convert Figma design tokens to CSS and Tailwind

bmartel opened the pull request:
This PR introduces a NX command to the `web` monorepo which allows for the processing of exported design tokens from our design system in Figma, and consumption in the application through SCSS and Tailwind config.

**Command:**
```
yarn design-tokens
```

Check the added README.md for more details.

<img width="773" alt="Screenshot 2025-03-18 at 5 59 14 AM" src="https://github.com/user-attachments/assets/244e7e23-2596-4a08-b947-531a2574f854" />

bmartel committed:
feat: OPTIC-1632: Convert Figma design tokens to CSS and Tailwind

bmartel committed:
move to proper tools location so it can be used from an nx command

bmartel committed:
refactor design-tokens-converter to work like an NX command

bmartel committed:
rename

bmartel committed:
fix the script variables passing

bmartel committed:
first working version of the output of colors

bmartel committed:
handle primitives in tailwind correctly

bmartel committed:
removing unused destination files

bmartel committed:
fix color token css generation

bmartel committed:
wip

bmartel committed:
check for the corrected key of corner-radius

bmartel committed:
wip other token handling

bmartel committed:
fix css token gen for spacing

bmartel committed:
end of file newline

bmartel committed:
fix tokens js gen

bmartel committed:
fix nested handling of primitive tokens in tailwind generation

bmartel committed:
consume tokens in app

bmartel committed:
incorporate design tokens into tailwind config

bmartel committed:
convert all px to rem

bmartel commented on web/tools/design-tokens-converter/README.md:
```diff
@@ -0,0 +1,143 @@
+  - Colors (light and dark modes)
+  - Spacing
+  - Typography (font family, font size, font weight, line height, letter spacing)
+  - Corner Radius (fixed from "corder-radius" typo in the source)
```
```suggestion
  - Corner Radius
```

bmartel committed:
Update web/tools/design-tokens-converter/README.md

bmartel committed:
linting

bmartel committed:
Merge branch 'fb-optic-1632' of github.com:HumanSignal/label-studio into fb-optic-1632

bmartel committed:
round the rem values to 4 decimal places

bmartel committed:
linting

bmartel committed:
zero is a unitless value

bmartel commented on web/tools/version/version:
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env node
 
-import './version.mjs';
+import "./version.mjs";
```
```suggestion
import './version.mjs';
```

bmartel committed:
Update web/tools/version/version

yyassi-heartex commented on web/design-tokens.json:
```diff
@@ -0,0 +1,5683 @@
+{
```
is this file generated from figma?

bmartel replied:
Yes, it is generated with the only plugin that works for dev mode accounts, and it makes me a bit nervous about the longevity and availability of these Figma plugins, so I'm opting to commit everything to ensure we aren't interrupted by anything that happens to change there over time.

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13924061595)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1632'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13924061595

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13924061595

nick-skriabin commented on web/libs/ui/src/tailwind.config.js:
```diff
@@ -190,6 +209,9 @@ module.exports = {
         md: "calc(var(--radius) - 2px)",
         sm: "calc(var(--radius) - 4px)",
+
+        // Add all border radii from tokens
```
```suggestion
        // Add all border radius from tokens
```

bmartel replied:
radii is the plural form here, otherwise it reads weird using the singular `radius` term.
